### PR TITLE
Presets functionality upgrade

### PR DESCRIPTION
**Pull request by portnov**
This adds:

* Support for "Standard" presets, which are supposed to be distributed with Sverchok (some kind of example settings). 
* Support for "Per-node presets", i.e. presets which store settings of one and only one node. Such presets are displayed in the node's N panel, so when the node is selected it is possible to switch it between several pre-saved settings. Some node classes already had similar functionality (I remember Spiral and Regular Solid).

"Per-node" presets are saved in special preset categories, which are distinguished with "/Node/" title prefix. For example, category "/Node/ Box" can only store presets for the "Box" node. Such categories are created automatically when you first create a preset for the certain type of node from the node's N panel. It is possible to use the usual Presets panel under the T panel to manage such presets.

Screenshot_20200510_104238

Also the list of all presets defined for the node is available in the node tree editor by *Shift-P* shortcut.

Screenshot_20200510_104511

## Difference between standard presets and Examples

Historically, our Examples menu contained examples of more or less complete setups, consisting of several nodes. Some of those examples are relatively complex. So, Examples is for examples of what can one do with Sverchok.

Per-node Presets, which are to be distributed with Sverchok, are just some combinations of settings of specific complex node, which developer thinks will be frequently used.

Standard Presets that are not per-node may be some small parts of logic that developer thinks may be frequently used. Probably it worth wrapping such pieces into Monads before packing as presets. As an example of such preset, I included Lloyd 2D monad.

## Preflight checklist

Put an x letter in each brackets when you're done this item:

- [x] Code changes complete.
- [x] Code documentation complete.
- [x] Documentation for users complete (or not required, if user never sees these changes).
- [x] Manual testing done. 
- [ ] Unit-tests implemented.
- [x] Ready for merge.

### CODE REVIEW

File: utils/sv_IO_panel_tools.py [hunk 1]
```diff
@@ -283,7 +283,11 @@ def create_dict_of_tree(ng, skip_set={}, selected=False, identified_node=None):
 
         IsMonadInstanceNode = (node.bl_idname.startswith('SvGroupNodeMonad'))
 
-        for k, v in node.items():
+        if save_defaults:
+            node_props = ((k, getattr(node, k)) for k in node.bl_rna.__annotations__.keys())
```
Comment (zeffii) [inline utils/sv_IO_panel_tools.py:287]: can i ask you to turn this into a function, which skips over pointers.
```python
node_props = get_node_annotations(node, pointers=False)
```
if this is to be used by nodes in general

Comment (portnov) [reply utils/sv_IO_panel_tools.py:287]: Ok

**Comment by vicdoval** [comment]
Great feature!! Wouldn't make more sense to place it in a different box layout just under the label? It seems a little strange to see it between "Edit Source" and "recreate node"

**Comment by portnov** [comment]
@vicdoval you mean move Presets block higher, above Help block?

**Comment by vicdoval** [comment]
Yes

**Comment by portnov** [comment]
No problem :)
Screenshot_20200510_011132

**Comment by portnov** [comment]
Well, I think I'll better replace this list of buttons with a menu box. Otherwise the panel will grow too big when there will be a lot of presets...
Screenshot_20200510_104238

Or maybe make it an option in settings?

Also it is possible to create a menu with list of presets for active node, which pops up by a hotkey (Shift-P, for example).

Screenshot_20200510_104511

What do you think?

**Comment by zeffii** [comment]
i think a popup is more preferable per node, than any additions of lists/menus in the NodeITem N panel menu.

**Pull request merged**